core/src/main/scala/doric/types/CustomSparkType.scala

```diff
@@ -0,0 +1,21 @@
+package doric.types
+
+import java.sql.{Date, Timestamp}
+import java.time.{Instant, LocalDate}
+
+trait CustomSparkType[T] {
+  type OriginalSparkType
+  val originalSparkType: SparkType[OriginalSparkType]
+  val literalTo: T => OriginalSparkType
+  val transform: OriginalSparkType => T
+}
+
+object CustomSparkType {
+  def apply[T: CustomSparkType]: CustomSparkType[T] = implicitly
+
+  implicit val fromDate: CustomSparkType[Date] =
```

Comment:
what is this for?

Reply:
It provides an evidence that the element is a doric custom type. It contain all the logic to transform from and to the original spark type. We also provide a implicit transformation to create a spark type with this class

Reply:
what I'm asking is why you added fromDate implicit in a custom type, why date?

Reply:
ah, natively we are going to accept only java 8 date API, so to make it compatible with previous java.sql API, we expose as a custom type those